### DISCUSSION TITLE
T-We from Telenor Norway

### DISCUSSION THREAD

**Discussion by KingKongKent** [fabianishere/udm-iptv#59] [Q&A]
T-we from Telenor does not use Vlan and I tried to set it up with the following config, but no success so far.
Not sure what i am doing wrong, do I need to add any FW rules? 

Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 224.0.0.0/4  93.91.111.0/24  148.122.0.0/16  239.0.0.0/8  233.0.0.0/8
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -0 staticroute -V IPTV_RG
LAN Interfaces: br0
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
4: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq qlen 1000
    inet 80.213.165.171/22 scope global dynamic eth8
       valid_lft 861sec preferred_lft 861sec
80.213.164.0/22 dev eth8  src 80.213.165.171
=== Service Logs ===
Feb 04 17:17:13 ubnt udm-iptv[58]: RECV V2 member report   from 192.168.1.99    to 239.255.255.254
Feb 04 17:17:13 ubnt udm-iptv[58]: No interfaces found for source 192.168.1.99
Feb 04 17:17:14 ubnt udm-iptv[58]: The IGMP message was local multicast. Ignoring.
Feb 04 17:17:14 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.79       to 224.0.0.251
Feb 04 17:17:14 ubnt udm-iptv[58]: Updated route entry for 224.0.0.251 on VIF #1
Feb 04 17:17:14 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.198      to 224.0.0.251
Feb 04 17:17:14 ubnt udm-iptv[58]: Updated route entry for 224.0.0.251 on VIF #1
Feb 04 17:17:15 ubnt udm-iptv[58]: The IGMP message was local multicast. Ignoring.
Feb 04 17:17:15 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.149      to 224.0.0.251
Feb 04 17:17:15 ubnt udm-iptv[58]: Updated route entry for 224.0.0.251 on VIF #1

A grep from the debug: (10.0.0.244 IPTV)
Feb 04 17:19:48 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.122
Feb 04 17:19:49 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94
Feb 04 17:20:06 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94
Feb 04 17:20:45 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94
Feb 04 17:20:53 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.122
Feb 04 17:20:59 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94
Feb 04 17:21:05 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94
Feb 04 17:22:05 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.122
Feb 04 17:22:06 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94
Feb 04 17:22:34 ubnt udm-iptv[58]: RECV V2 member report   from 10.0.0.244      to 233.184.48.94

**Comment by fabianishere** [comment]
I suggest putting the IPTV decoders on a separate VLAN.

**Comment by KingKongKent** [comment]
Thanks for your quick response:
Moved the Ip decoder to a Vlan:
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 224.0.0.0/4  93.91.111.0/24  148.122.0.0/16  239.0.0.0/8  233.0.0.0/8
WAN VLAN: 2 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br2
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Feb 04 17:47:50 ubnt udm-iptv[4435]: udhcpc failed to get a DHCP lease
Feb 04 17:47:50 ubnt udm-iptv[4435]: No lease, forking to background
Feb 04 17:47:50 ubnt udm-iptv[4435]: NATing IPTV network ranges (if necessary)
Feb 04 17:47:50 ubnt udm-iptv[4435]: Setting up igmpproxy
Feb 04 17:47:50 ubnt udm-iptv[4435]: Starting igmpproxy
Feb 04 17:47:50 ubnt udm-iptv[4435]: adding VIF, Ix 0 Fl 0x0 IP 0x0102a8c0 br2, Threshold: 1, Ratelimit: 0
Feb 04 17:47:50 ubnt udm-iptv[4435]: There must be at least 1 Vif as upstream.
Feb 04 17:47:50 ubnt systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Feb 04 17:47:50 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Feb 04 17:47:50 ubnt systemd[1]: udm-iptv.service: Consumed 111ms CPU time.

# ip route list
10.0.0.0/24 dev br0 proto kernel scope link src 10.0.0.1
80.213.164.0/22 dev eth8 proto kernel scope link src 80.213.165.171
192.168.2.0/24 dev br2 proto kernel scope link src 192.168.2.1

Debug:

Feb 04 17:52:02 ubnt udm-iptv[12117]: Device iptv already exists.. Deleting device
Feb 04 17:52:02 ubnt udm-iptv[12117]: Obtaining IP address for VLAN interface
Feb 04 17:52:02 ubnt udm-iptv[12117]: udhcpc (v1.22.1) started
Feb 04 17:52:02 ubnt udm-iptv[12117]: Sending discover...
Feb 04 17:52:05 ubnt udm-iptv[12117]: Sending discover...
Feb 04 17:52:08 ubnt udm-iptv[12117]: Sending discover...
Feb 04 17:52:11 ubnt udm-iptv[12117]: udhcpc failed to get a DHCP lease
Feb 04 17:52:11 ubnt udm-iptv[12117]: No lease, forking to background
Feb 04 17:52:11 ubnt udm-iptv[12117]: NATing IPTV network ranges (if necessary)
Feb 04 17:52:11 ubnt udm-iptv[12117]: Setting up igmpproxy
Feb 04 17:52:11 ubnt udm-iptv[12117]: Starting igmpproxy
Feb 04 17:52:11 ubnt udm-iptv[12117]: adding VIF, Ix 0 Fl 0x0 IP 0x0102a8c0 br2, Threshold: 1, Ratelimit: 0
Feb 04 17:52:11 ubnt udm-iptv[12117]: There must be at least 1 Vif as upstream.
Feb 04 17:52:11 ubnt systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXC
EPTION
Feb 04 17:52:11 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Feb 04 17:52:11 ubnt systemd[1]: udm-iptv.service: Consumed 120ms CPU time.
Feb 04 17:52:16 ubnt systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Feb 04 17:52:16 ubnt systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 26.
Feb 04 17:52:16 ubnt systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Feb 04 17:52:16 ubnt systemd[1]: udm-iptv.service: Consumed 120ms CPU time.
Feb 04 17:52:16 ubnt systemd[1]: Started IPTV support for the UniFi Dream Machine.
Feb 04 17:52:16 ubnt udm-iptv[12523]: Device iptv already exists.. Deleting device
Feb 04 17:52:16 ubnt udm-iptv[12523]: Obtaining IP address for VLAN interface
Feb 04 17:52:16 ubnt udm-iptv[12523]: udhcpc (v1.22.1) started
Feb 04 17:52:16 ubnt udm-iptv[12523]: Sending discover...
Feb 04 17:52:19 ubnt udm-iptv[12523]: Sending discover...
Feb 04 17:52:22 ubnt udm-iptv[12523]: Sending discover...
Feb 04 17:52:25 ubnt udm-iptv[12523]: udhcpc failed to get a DHCP lease
Feb 04 17:52:25 ubnt udm-iptv[12523]: No lease, forking to background
Feb 04 17:52:25 ubnt udm-iptv[12523]: NATing IPTV network ranges (if necessary)
Feb 04 17:52:25 ubnt udm-iptv[12523]: Setting up igmpproxy
Feb 04 17:52:25 ubnt udm-iptv[12523]: Starting igmpproxy
Feb 04 17:52:25 ubnt udm-iptv[12523]: adding VIF, Ix 0 Fl 0x0 IP 0x0102a8c0 br2, Threshold: 1, Ratelimit: 0
Feb 04 17:52:25 ubnt udm-iptv[12523]: There must be at least 1 Vif as upstream.
Feb 04 17:52:25 ubnt systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXC
EPTION
Feb 04 17:52:25 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Feb 04 17:52:25 ubnt systemd[1]: udm-iptv.service: Consumed 128ms CPU time.
Feb 04 17:52:31 ubnt systemd[1]: udm-iptv.service: Service RestartSec=5s expired, scheduling restart.
Feb 04 17:52:31 ubnt systemd[1]: udm-iptv.service: Scheduled restart job, restart counter is at 27.
Feb 04 17:52:31 ubnt systemd[1]: Stopped IPTV support for the UniFi Dream Machine.
Feb 04 17:52:31 ubnt systemd[1]: udm-iptv.service: Consumed 128ms CPU time.
Feb 04 17:52:31 ubnt systemd[1]: Started IPTV support for the UniFi Dream Machine.
Feb 04 17:52:31 ubnt udm-iptv[12931]: Device iptv already exists.. Deleting device
Feb 04 17:52:31 ubnt udm-iptv[12931]: Obtaining IP address for VLAN interface
Feb 04 17:52:31 ubnt udm-iptv[12931]: udhcpc (v1.22.1) started
Feb 04 17:52:31 ubnt udm-iptv[12931]: Sending discover...
Feb 04 17:52:34 ubnt udm-iptv[12931]: Sending discover...

**Reply by fabianishere** [reply]
WAN VLAN should not be changed in that case.

**Reply by KingKongKent** [reply]
This should be more correct than,..
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 224.0.0.0/4  93.91.111.0/24  148.122.0.0/16  239.0.0.0/8  233.0.0.0/8
WAN VLAN: 0 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br2
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
4: eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq qlen 1000
    inet 80.213.165.171/22 scope global dynamic eth8
       valid_lft 1151sec preferred_lft 1151sec
80.213.164.0/22 dev eth8  src 80.213.165.171
=== Service Logs ===
Feb 04 18:22:50 ubnt udm-iptv[32033]: Updated route entry for 233.184.48.122 on VIF #1
Feb 04 18:22:50 ubnt udm-iptv[32033]: Adding MFC: 93.91.111.2 -> 233.184.48.122, InpVIf: 0
Feb 04 18:22:50 ubnt udm-iptv[32033]: Joining group 233.184.48.122 on interface eth8
Feb 04 18:22:50 ubnt udm-iptv[32033]: Route activation request from 80.213.165.171 for 233.89.188.1 is from myself. Ignoring.
Feb 04 18:22:51 ubnt udm-iptv[32033]: RECV V2 member report   from 192.168.2.6     to 233.184.48.94
Feb 04 18:22:51 ubnt udm-iptv[32033]: Updated route entry for 233.184.48.94 on VIF #1
Feb 04 18:22:51 ubnt udm-iptv[32033]: Adding MFC: 93.91.111.2 -> 233.184.48.94, InpVIf: 0
Feb 04 18:22:51 ubnt udm-iptv[32033]: Joining group 233.184.48.94 on interface eth8
Feb 04 18:22:53 ubnt udm-iptv[32033]: RECV V2 member report   from 192.168.2.1     to 224.0.0.251
Feb 04 18:22:53 ubnt udm-iptv[32033]: The IGMP message was from myself. Ignoring.

**Comment by KingKongKent** [comment]
I got it to work with creating a FW rule that allow UPD from Internet to VLAN with source 233.0.0.0/8, 148.122.0.0/16, 224.0.0.0/4, 93.91.111.0/24, 239.0.0.0/8
